services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/DefaultSqsBatchManager.java

```diff
@@ -54,18 +60,17 @@
     private final BatchManager<ChangeMessageVisibilityRequest, ChangeMessageVisibilityResponse,
         ChangeMessageVisibilityBatchResponse> changeVisibilityBatchManager;
 
+    // TODO: When generating executor, do it like it is done here in
     private DefaultSqsBatchManager(DefaultBuilder builder) {
         this.client = builder.client;
         SqsBatchConfiguration config = new SqsBatchConfiguration(builder.overrideConfiguration);
         BatchOverrideConfiguration overrideConfiguration = BatchOverrideConfiguration.builder()
                                                                                      .maxBatchItems(config.maxBatchItems())
                                                                                      .maxBatchOpenInMs(config.maxBatchOpenInMs())
                                                                                      .build();
-
-
         ScheduledExecutorService scheduledExecutor = builder.scheduledExecutor;
-        this.executor = builder.executor;
-
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().threadNamePrefix("DefaultSqsBatchManager").build();
+        this.executor = createDefaultExecutor(threadFactory);
```

Comment:
We probably should only create a default executor if `builder.executor` is null and only close the executor if it's the default one created by the class

Reply:
Only creating and shutting down default executor if builder.executor is null. Adding tests to check if executors are properly shut down.